### ISSUE TITLE
Create layer extension and endpoints

### ISSUE THREAD

**Issue by jisantuc** [azavea/franklin#599]
`stac4s` defines a `layer` extension, effectively for ad hoc groupings of items into catalogs that are orthogonal to what collections the items live in. Basically you use layers to say "I like this item, and I like this item, and I'd like a way to put them in the same container without running afoul of the one-collection rule."

Franklin has already been used to store and serve items that support this extension. However, working with layers with the existing endpoints isn't extremely convenient -- to add an item to a layer, you have to send a `PATCH`. To find items in a layer, you have to use the query extension. To create a layer... well you just can't create a layer. Sad times.

Here are the endpoints that I'll propose:

```
* /layers GET -- list all the layers
* /layers POST<layer json> -- create a layer
* /layers/<id> GET -- get a specific layer
* /layers/<id> DELETE -- delete a layer, remove references to it from any items that refer to it
* /layers/<id>/items GET -- list all the items in a layer
* /layers/<id>/items POST<list of ids> -- add one item to a layer
* /layers/<id>/items PUT<list of ids> -- replace items in this layer
* /layers/<id>/items DELETE -- remove all items from this layer
* /layers/<id>/items/<item-id> GET -- get an item in this layer (404s on item-not-in-layer)
* /layers/<id>/items/<item-id> DELETE -- remove this item from this layer (404s on item-not-in-layer)
```

Serving the layer endpoints should be configurable with a startup flag `--with-layers`, which should be included in the docs.

The extension itself should be thrown with the `stac4s` layers extension into `azavea/stac-extension-layers`

**Comment by pomadchin** [comment]
Just a note: after endpoints would be implemented we need to reflect changes here: https://github.com/azavea/stac-layer 
(leaving this comment here, otherwise I will forget about it)

**Comment by jisantuc** [comment]
This was mostly implemented in #612, but is paused for now. We'll resume this and can reopen based on that PR if we discover we need this after all
